plugins/gradle/src/index.ts

```diff
@@ -112,12 +112,9 @@ export default class GradleReleasePluginPlugin implements IPlugin {
     if (buildFlag) {
       // don't create release, tag, or commit since auto will do this
       await execPromise(this.options.gradleCommand, [
-        "release",
```

Comment:
This code is the same as the other block in the conditional. At the very least we need to ensure that the `buildFlag` is respected and the projects [`buildTasks`](https://github.com/researchgate/gradle-release/blob/master/src/main/groovy/net/researchgate/release/ReleasePlugin.groovy#L85) is called to ensure that anything relying on the version string is re-built.

Reply:
I'm thinking that we'll probably also want to ensure that other features of the release plugin are still being utilized here, like [`checkSnapshotDependencies`](https://github.com/researchgate/gradle-release/blob/master/src/main/groovy/net/researchgate/release/ReleasePlugin.groovy#L82).

Reply:
@ykhandelwal913 This looks better. The only thing I see here that is a little confusing is the `createScmAdapter`, `initScmAdapter`, and `unSnapshotVersion` task calls. 

For both `createScmAdapter` and `initScmAdapter`, I can maybe see some issues here if those aren't initialized by some dependent task, but would like to know for sure if you saw an error without them.

For `unSnapshotVersion`, I think we should try to avoid this one if we can since this would actually change the version in the version file, which we don't want because we are already changing it ourselves with `updateVersion`.

Does that make sense?

Reply:
createScmAdapter and initScmAdapter are internal dependency to subsequent task. Hence added them. Removed the unSnapshot version